boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,180 @@
+import org.ow2.proactive.resourcemanager.utils.OperatingSystem;
+
+
+public class IAMStarter {
+
+    private static final Logger LOGGER = Logger.getLogger(IAMStarter.class);
+
+    private static final String OS = System.getProperty("os.name");
+
+    private static final String SEPARATOR = File.separator;
+
+    private static Configuration config = new BaseConfiguration();
+
+    private static Process process;
+
+    private static List<String> command = new ArrayList<>();
+
+    private static boolean started = false;
+
+    private IAMStarter() {
+
+    }
+
+    /**
+     *  Start IAM microservice
+     */
+    public static Process start(String paHome, String bootMicroservicesPath, String bootConfigurationPath)
+            throws InterruptedException, IOException, ExecutionException, ConfigurationException {
+
+        if (!started) {
+            LOGGER.info("Starting IAM microservice...");
+
+            // load IAM configuration
+            loadIAMConfiguration(bootConfigurationPath);
+
+            // build java command to launch IAM
+            buildJavaCommand(paHome);
+
+            // find IAM war archive path
+            buildMicroservicePath(bootMicroservicesPath);
+
+            // execute IAM as a separate JAva process
+            ProcessBuilder processBuilder = new ProcessBuilder(command);
+            processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+            process = processBuilder.start();
+            LOGGER.info(streamOutput(process.getInputStream()));
+
+            started = true;
+        }
+        return process;
+    }
+
+    /**
+     * Stream microservice output (using an executor) to check that it starts properly
+     */
+    private static String streamOutput(InputStream inputStream) throws InterruptedException, ExecutionException {
+
+        // Stream microservice output
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future<String> future = executor.submit((Callable) () -> {
+
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
+                String line;
+
+                while ((line = br.readLine()) != null) {
+                    if (line.contains(config.getString(IAMConfiguration.READY_MARKER))) {
+                        break;
+                    }
+                }
+            }
+            return "IAM Microservice started";
+        });
+
+        // Check for timeout
+        try {
+            return future.get(config.getInt(IAMConfiguration.STARTUP_TIMEOUT), TimeUnit.SECONDS);
+        } catch (TimeoutException toe) {
+            return "IAM Microservice timed out to start. See the logs for the details.";
+        } finally {
+            executor.shutdownNow();
+            future.cancel(true);
+        }
+    }
+
+    /**
+     * Prepare java command with jvm args
+     */
+    private static void buildJavaCommand(String paHome) {
```

Comment:
Isn't there a CommandBuilder class in scheduling? Can you check if it is reusable in this case?

Reply:
Effectively, there is a class called 'CommandLineBuilder', but I cannot use it as it is tailored to execute the class 'RMNodeStarter' and not to execute separate jars.